### DISCUSSION TITLE
Hubs Local Development JWT and setTimeout Issue

### DISCUSSION THREAD

**Discussion by ArdaKonut** [mozilla/hubs#6497] [General Questions & Discussion]
Hi everyone, I've been working on local hubs development project by following this tutorial: https://github.com/albirrkarim/mozilla-hubs-installation-detailed?tab=readme-ov-file
Right now, I have six servers (Hubs,Hubs/Admin,Spoke,dialog,Reticulum and postgREST) running on WSL. When I navigate and run the landing page (https://localhost:4000/), I can see an error message related with JWT Token, Jose and Guardian on Reticulum's IEX(SS is attached below). At the same time, two big errors appear on the browser's console. The first one is related to CSP and the second one is related with the runtime issue that I've mentioned above (As far as I understood) and it iterates itself continuously. Any idea or help would be appreciated since this is an important part of my masters degree graduation project. Thanks in advance.
![Ekran görüntüsü 2024-09-03 215944](https://github.com/user-attachments/assets/30c597e4-4e64-46df-94fb-ee2044ac57c5)
![Ekran görüntüsü 2024-09-03 220243](https://github.com/user-attachments/assets/bcdbe372-c40d-4053-90e3-4be599edc9cc)
![Ekran görüntüsü 2024-09-03 220419](https://github.com/user-attachments/assets/ff78f440-030a-42f7-a2f6-fefab079cfb9)

**Comment by Exairnous** [comment]
I'm not familiar with that way of installing/running Hubs, but have you created an admin user for Reticulum?  That might be the issue?  Not sure.

Are you aware of Hubs Compose?  It should be an easier way to run and develop Hubs locally.
https://github.com/Hubs-Foundation/hubs-compose/

**Reply by ArdaKonut** [reply]
Sorry for replying late. Yes, I've created an admin user for Reticulum. I think this issue is related with JOSE and Guardian libraries which handle JWT Token and conversion from RSA to JWK and these kind of details. I couldn't manage to get it sorted so right now I'm following your advice and move on with Hubs Compose. Thank you very much.

**Reply by Exairnous** [reply]
No worries, and you're welcome!  I hope Hubs Compose works well for you, but if you run into any trouble, feel free to ask questions in GitHub or the Hubs Discord :)
